cvat-data/src/js/cvat-data.js

```diff
@@ -141,96 +58,49 @@ class FrameProvider {
 
     */
 
-    startDecode(block, start, end, callback) {
-        if (this._running) {
-            const error = new Error('Decoding has already running');
-            const promise = new Promise((resolve, reject) => {
-                if (this._promise) {
-                    this._promise.reject();
-                }
-
-                this._promise = {
-                    resolve,
-                    reject,
-                };
-            });
-            error.donePromise = promise;
-            throw error;
-        }
+    startDecode(block, start, end, callback)
+    {
+         if (this._blockType === BlockType.TSVIDEO){
+            if (this._running) {
```

Comment:
Where is the "this._running" set to true?

Reply:
I think we can delete  "this._running" everywhere. Becouse we send all blocks and start/end to thread and it returns frames with id. We can run several threads in same time. I need to try it)

Reply:
@ygnn123

We can try to do so.

Reply:
check. It works for me. I send 2 blocks at same times and get 600 frames.